frontends/main/src/app-pages/DashboardPage/CoursewareDisplay/EnrollmentDisplay.tsx

```diff
@@ -178,24 +215,184 @@ const EnrollmentExpandCollapse: React.FC<EnrollmentExpandCollapseProps> = ({
+  )
+  const program = rawProgram ? mitxonlineProgram(rawProgram) : undefined
+  const { data: rawProgramCourses, isLoading: programCoursesLoading } =
+    useQuery(coursesQueries.coursesList({ id: program?.courseIds }))
+
+  // Build sections from requirement tree
+  const requirementSections =
+    program?.reqTree
+      .filter((node) => node.data.nodeType === "operator")
+      .map((node) => {
+        const courseIds = extractCoursesFromNode(node)
+        const sectionCourses = rawProgramCourses?.results
+          .filter((course) => courseIds.includes(course.id))
+          .map((course) => {
+            const enrollment = userCourses?.find((dashboardCourse) =>
+              course.courseruns.some(
+                (run) => run.courseware_id === dashboardCourse.coursewareId,
+              ),
+            )?.enrollment
+            return mitxonlineCourse(course, enrollment)
+          })
+
+        return {
+          title: getRequirementSectionTitle(node),
+          courses: sectionCourses || [],
+          node,
+        }
+      })
+      .filter((section) => section.courses.length > 0) || []
+
+  const allProgramCourses = requirementSections.flatMap(
+    (section) => section.courses,
+  )
+  const completedCount = allProgramCourses.filter(
+    (course) => course.enrollment?.status === EnrollmentStatus.Completed,
+  ).length
+  const totalCount = allProgramCourses.length
+  return (
+    <>
+      <Stack direction="column">
+        <Stack direction="column" marginBottom="24px">
+          <Typography variant="h5" color={theme.custom.colors.silverGrayDark}>
+            MITx | {program?.programType}
+          </Typography>
+          <Typography variant="h3" paddingBottom="32px">
+            {program?.title}
+          </Typography>
+          <Typography variant="body2">
+            You have completed
+            <Typography component="span" variant="subtitle2">
+              {" "}
+              {completedCount} of {totalCount} courses{" "}
+            </Typography>
+            for this program.
+          </Typography>
+        </Stack>
+        {requirementSections.map((section, index) => {
+          const sectionCompletedCount = section.courses.filter(
+            (course) =>
+              course.enrollment?.status === EnrollmentStatus.Completed,
+          ).length
+
+          return (
+            <React.Fragment key={index}>
+              <Stack
+                direction="row"
+                justifyContent="space-between"
+                marginBottom="16px"
+                marginTop={index > 0 ? "32px" : "0"}
+              >
+                <Typography variant="subtitle2" color={theme.custom.colors.red}>
+                  {section.title}
+                </Typography>
+                <Typography
+                  variant="body2"
+                  color={theme.custom.colors.silverGrayDark}
+                >
+                  Completed {sectionCompletedCount} of {section.courses.length}
```

Comment:
I don't think this display is quite right for electives nodes.

My program has:
```
{
  "data": {
    "node_type": "operator",
    "operator": "min_number_of",
    "operator_value": "1",
    "program": 1,
    "course": null,
    "required_program": null,
    "title": "Electives!",
    "elective_flag": true
  },
  "id": 18,
  "children": [/* two child course nodes */]
}
```
I think it should display like this (note "0 of 1" not "0 of 2", since you only need to take 1 elective).

<img width="776" height="237" alt="Image" src="https://github.com/user-attachments/assets/19d60ce6-255f-4d56-9b2e-52660225468a" />